docs/analysis/competing-tools-detection.md

```
@@ -0,0 +1,206 @@
+- `title`: Window title (from Accessibility framework `AXUIElement`)
+- Bundle identifier (for browser detection: Chrome, Firefox, Safari variants)
+
+### Platform Strategy Pattern
+
+ActivityWatch uses a **strategy pattern** for platform abstraction:
+- `macos-swift`: Compiled Swift binary using NSWorkspace + AXUIElement
+- `linux-x11`: Uses `xprop` and X11 APIs
+- `linux-wayland`: Uses wlr-foreign-toplevel-management
+- `windows`: Uses Win32 API
+
+### Lessons for Us
```

markdownlint-cli2 (0.20.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)